app/models/organization.rb

```diff
@@ -93,4 +92,11 @@ def root_org_setting(setting)
   def self.descendants
     ObjectSpace.each_object(Class).select { |klass| klass < self }
   end
+
+  def get_time_zone
+    org = self.self_and_ancestors.where.not(time_zone: '').last
+    return Time.zone unless org
+    return org.setting("time_zone")
```

Comment:
I typically write this as 
```suggestion
if (org = self.self_and_ancestors.where.not(time_zone: '').or(where.not(time_zone: nil)).last)
  org.setting("time_zone")
else
  TIme.zone
end
```